### PR TITLE
Wrappers and cleanups

### LINKED ISSUE #5: Convenient wrapper for WasiCli0_2_0_EnvironmentGetArguments and WasiCli0_2_0_EnvironmentGetEnvironment

### LINKED ISSUE #6: Create wrappers for golem's host functions

### PR DESCRIPTION
**Pull request by noise64**
Resolves: https://github.com/golemcloud/golem-go/issues/5

 - golem host API wrappers for:
   - idempotence mode
   - persistence level
   - retry policy
  - cli API wrappers for:
    - args
    - env vars
 - cleanups for RoundTripper
 - added test main app and makefile (no CI for now)
 - moved generated binding under the `binding` package name and remove (git ignored) binding.c to prevent duplicated symbols in component projects
 - convenience methods for initialize standard lib

### CODE REVIEW

File: golemhost/retrypolicy.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	"time"
+
+	binding "github.com/golemcloud/golem-go/golem_go_bindings"
+)
+
+type RetryPolicy struct {
+	MaxAttempts uint32
+	MinDelay    time.Duration
+	MaxDelay    time.Duration
+	Multiplier  uint32
```
Comment (vigoo) [inline golemhost/retrypolicy.go]: This should have jitter too now

Comment (vigoo) [reply golemhost/retrypolicy.go]: But it's not in the WIT definition yet. However multiplier is now `f64`:
https://github.com/golemcloud/golem-wit/blob/main/wit/deps/golem/golem-host.wit#L49

File: net/http/roundtrip.go [hunk 2]
```diff
@@ -8,49 +8,50 @@ import (
 	"strconv"
 	"strings"
 
-	golem_go_bindings "github.com/golemcloud/golem-go/golem_go_bindings"
```
Comment (vigoo) [inline net/http/roundtrip.go:11]: Please update the go test component in the `golem` repo which uses this after we merge this

**Review by vigoo — APPROVED** [review]
After updated to the latest WIT, good to go

File: net/http/roundtrip.go (hunk 2 above)
Comment (noise64) [reply net/http/roundtrip.go:11]: https://github.com/golemcloud/golem-go/pull/8/commits/61513510b90b0f2fa6d9cc42044509322b0a5501
https://github.com/golemcloud/golem-go/pull/8/commits/5901d4fcd50513a1cfd147696cded8ffe62e3cf1

**Pull request merged**